experimental/add_to_app/android_fullscreen/app/src/main/java/dev/flutter/example/androidfullscreen/MainActivity.kt

```diff
@@ -6,18 +6,71 @@ package dev.flutter.example.androidfullscreen
 
 import android.os.Bundle
 import android.widget.Button
+import android.widget.TextView
 import androidx.appcompat.app.AppCompatActivity
 import io.flutter.embedding.android.FlutterActivity
+import io.flutter.embedding.engine.FlutterEngine
+import io.flutter.embedding.engine.FlutterEngineCache
+import io.flutter.embedding.engine.dart.DartExecutor
+import io.flutter.plugin.common.MethodChannel
+import io.flutter.view.FlutterMain
 
 class MainActivity : AppCompatActivity() {
 
+    private val ENGINE_ID = "1"
+
+    private var channel : MethodChannel? = null
+    private var count = 0
+    private var counterLabel : TextView? = null
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
 
+        FlutterMain.ensureInitializationComplete(applicationContext, null)
+
+        // Create and warm up Flutter engine.
+        val flutterEngine = FlutterEngine(this)
+
+        flutterEngine
+            .dartExecutor
+            .executeDartEntrypoint(
+                DartExecutor.DartEntrypoint.createDefault()
+            )
+
+        FlutterEngineCache.getInstance().put(ENGINE_ID, flutterEngine)
```

Comment:
This caching mechanism is probably best demonstrated in an `Application` subclass. Generally, one would like to avoid having `ActivityA` know any meaningful details about `ActivityB`. So i'd be a little worried about people replicating this approach in their own app.

Reply:
Got it. I moved the caching and channel setup into the application class, and I'm using a simple observable to propagate data out to my main activity. Let me know what you think.

Another option would be to subclass FlutterActivity, have it manage the channel and return a result, and update MainActivity to use startActivityForResult. I got an hour into implementing it that way, then hit a wall when I didn't know where to cut in and set the result before the FlutterActivity Destroyed itself.